source/s3_request_messages.c

```diff
@@ -754,11 +756,29 @@ struct aws_input_stream *aws_s3_message_util_assign_body(
         if (checksum_config->location == AWS_SCL_TRAILER) {
             /* aws-chunked encode the payload and add related headers */
 
-            /* set Content-Encoding header. TODO: the aws-chunked should be appended to the existing content encoding.
+            /* set Content-Encoding header. If the header already exists, append the exisiting value to aws-chunked
              */
-            if (aws_http_headers_set(headers, g_content_encoding_header_name, g_content_encoding_header_aws_chunked)) {
+            struct aws_byte_cursor content_encoding_header_cursor;
```

Comment:
Add a comment that we checked that the existing value cannot be `aws-chunked` [here](https://github.com/awslabs/aws-c-s3/blob/main/source/s3_client.c#L744-L760). 

And, debatable: is it worth to create a meta-request level of content-encoding header, so that we don't need to do this append every time when we create a single request?

Reply:
Thanks, I have updated the comment. I thought about adding it at meta-request level but this value is only used in one particular case which makes it a bit confusing to add it at the meta request level.